File: Javascript/0851-0900/881. Boats to Save People.js
// https://leetcode.com/problems/boats-to-save-people/

var numRescueBoats = function(people, limit) {
    people.sort((a, b) => a - b);

    let first = 0, second = people.length-1, result = 0;

    while(first <= second){
        let weight = people[first] + people[second];
        
        first += (weight <= limit) ? 1 : 0;
        second--; result++;
    }

    return result;
};

File: Python/0851-0900/881. Boats to Save People.py
# https://leetcode.com/problems/boats-to-save-people/

class Solution(object):
    def numRescueBoats(self, people, limit):
        people.sort()

        first, second, result = 0, len(people)-1, 0

        while first <= second:
            weight = people[first] + people[second]
            
            first += 1 if (weight <= limit) else 0
            second -= 1
            result += 1

        return result
        
